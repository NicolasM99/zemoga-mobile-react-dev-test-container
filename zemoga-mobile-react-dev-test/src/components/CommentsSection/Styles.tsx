import { StyleSheet } from 'react-native';

export const commentsSectionStyles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
    position: 'absolute',
    height: '100%',
    width: '100%'
  },
  content: {
    flex: 1,
    backgroundColor: 'white',
    borderRadius: 20,
    padding: 16,
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: -2
    },
    shadowOpacity: 0.25,
    shadowRadius: 16,
    elevation: 5,
    maxHeight: '90%'
  },
  closeBtn: {
    flex: 1,
    position: 'absolute',
    top: -16,
    right: -16,
    padding: 16
  }
});
