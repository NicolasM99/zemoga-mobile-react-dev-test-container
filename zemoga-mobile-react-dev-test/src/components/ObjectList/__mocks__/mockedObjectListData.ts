export const mockedObjectList = {
  test: 'Test attribute',
  nested0: {
    nestedlvl0attribute: 'Nested level 0 attribute',
    nested1: {
      nestedlvl1attribute: 'Nested level 1 attribute'
    }
  }
};
