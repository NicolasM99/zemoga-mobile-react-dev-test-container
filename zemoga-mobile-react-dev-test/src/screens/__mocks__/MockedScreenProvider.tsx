import React from 'react';

import { NavigationContainer } from '@react-navigation/native';

import { PostsListProvider } from 'src/components/PostsList/context/PostsListContext';
import { InternetStatusProvider } from 'src/context/InternetStatusContext';

interface MockedScreenProviderProps {
  children: React.ReactNode;
}

const MockedScreenProvider: React.FC<MockedScreenProviderProps> = ({
  children
}: MockedScreenProviderProps) => (
  <NavigationContainer>
    <InternetStatusProvider>
      <PostsListProvider>{children}</PostsListProvider>
    </InternetStatusProvider>
  </NavigationContainer>
);

export default MockedScreenProvider;
