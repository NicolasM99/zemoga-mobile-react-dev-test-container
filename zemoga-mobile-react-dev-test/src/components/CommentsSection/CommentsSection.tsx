import React, { FC, useImperativeHandle, useCallback } from 'react';

import { FontAwesome5 } from '@expo/vector-icons';
import { View, Modal, TouchableOpacity, FlatList } from 'react-native';

import { ICommentsSection } from './@types/commentsSection';
import Comment from './components/Comment/Comment';
import { commentsSectionStyles } from './Styles';
import Header from '../Header/Header';

const CommentsSection: FC<ICommentsSection> = ({ comments = [], modalRef }: ICommentsSection) => {
  const [openModal, setOpenModal] = React.useState(false);

  useImperativeHandle(modalRef, () => ({
    closeModal: () => setOpenModal(false),
    openModal: () => setOpenModal(true)
  }));

  const handleCloseModal = () => {
    setOpenModal(false);
  };
  const CloseBtn = () => (
    <TouchableOpacity onPress={() => handleCloseModal()} style={commentsSectionStyles.closeBtn}>
      <FontAwesome5 name="times" size={20} />
    </TouchableOpacity>
  );
  const renderItem = useCallback(
    ({ item: { id, ...rest } }: any) => <Comment key={id} id={id} {...rest} />,
    []
  );
  return (
    <Modal testID="modalComments" animationType="slide" transparent={true} visible={openModal}>
      <View style={commentsSectionStyles.container}>
        <View style={commentsSectionStyles.content}>
          <CloseBtn />
          <Header>Comments</Header>
          <FlatList
            data={comments}
            renderItem={renderItem}
            initialNumToRender={10}
            maxToRenderPerBatch={10}
            removeClippedSubviews
            windowSize={10}
          />
        </View>
      </View>
    </Modal>
  );
};

export default CommentsSection;
