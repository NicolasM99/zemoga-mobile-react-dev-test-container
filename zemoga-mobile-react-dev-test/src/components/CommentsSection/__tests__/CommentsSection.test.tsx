import React from 'react';

import { render, screen } from '@testing-library/react-native';

import { mockedComments } from '../__mocks__/mockedCommentsSectionData';
import CommentsSection from '../CommentsSection';

describe('<CommentsSection /> unit tests', () => {
  it('Should mount correctly the component', () => {
    render(<CommentsSection comments={[]} />);
    expect(screen).toMatchSnapshot();
  });

  it('Should mount correctly the component and open the modal (visible prop true)', () => {
    const realUseState = React.useState as any;
    const stubInitialState = [true];
    jest.spyOn(React, 'useState').mockImplementationOnce(() => realUseState(stubInitialState));
    render(<CommentsSection comments={mockedComments} modalRef={null} />);
    expect(screen.getByTestId('modalComments').props.visible[0]).toBeTruthy();
  });
});
