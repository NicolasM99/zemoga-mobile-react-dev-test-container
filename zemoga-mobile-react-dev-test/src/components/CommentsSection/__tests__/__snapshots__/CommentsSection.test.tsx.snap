// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentsSection /> unit tests Should mount correctly the component 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  testID="modalComments"
  transparent={true}
  visible={false}
/>
`;
